### PR TITLE
Note Schema updated

### LINKED ISSUE #15: enable attachments (image, video, etc.)
I am able to add attachment features to this project on both the backend and frontend.

### PR DESCRIPTION
**Pull request by saurabh-dev-vns**
## Description

Please include a summary of the change and which issue is fixed. Also include relevant motivation and context.

- Fixes #15 

## Type of change

Please delete options that are not relevant.

- [done] Added an attachments field to store the paths of uploaded files while making notes.
- [done] The frontend has been updated to allow file uploads.
- [done] Add an attachment field to upload files directly to the upload folder, and send the file path to the backend to store the uploaded file's path.
- [pending] Edit the attachment.
- [pending] Delete the uploaded attachment data from the upload folder and remove its path from the database.

## Checklist

- [done ] My code follows the style guidelines of this project
- [ done] I have performed a self-review of my own code
- [done ] I have commented my code, particularly in hard-to-understand areas
- [dene] My changes generate no new warnings
- [done ] New and existing unit tests pass locally with my changes
- [done ] I have maintained a clean commit history by using the necessary Git commands
- [done ] I have checked that my code does not cause any merge conflicts

## Screenshots (if applicable)

Add screenshots to help explain the changes (if necessary).

### CODE REVIEW

**Comment by saurabh-dev-vns** [comment]
Hey @theashikagupta, @yashmandi, please approve this PR. I have made some changes to resolve issue #15.

**Comment by saurabh-dev-vns** [comment]
Hey @theashikagupta, @yashmandi, please approve this PR. I have made some changes to resolve issue #15

**Comment by theashikagupta** [comment]
Add screen recording

File: frontend/src/components/Input/AddAttachmentInput.jsx [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  const fileInputRef = useRef(null); // Create a ref for the file input
+
+  const handleFileChange = (event) => {
+    const selectedFiles = Array.from(event.target.files); // Get the selected files
+    setFiles(selectedFiles); // Update the state with the selected files
+  };
+
+  const handleButtonClick = () => {
+    if (files.length > 0) {
+      console.log('Files selected:', files);
```
Comment (Abhishek332) [inline frontend/src/components/Input/AddAttachmentInput.jsx:15]: we should avoid console logs in production

Comment (saurabh-dev-vns) [reply frontend/src/components/Input/AddAttachmentInput.jsx:15]: I'll take care of this next time.

**Comment by saurabh-dev-vns** [comment]
I have a screen recording of the file upload attachment, but I haven't added the functionality to view attachments in the notes yet. I will implement that later. For now, please accept my PR, @theashikagupta, @yashmandi.

https://github.com/user-attachments/assets/c0f01995-61f1-4845-a1fb-f5b9f8471b7f

**Comment by saurabh-dev-vns** [comment]
Why isn't my merged effect reflecting on the site, @yashmandi ?

**Comment by theashikagupta** [comment]
Check once now it will reflect . There were some issues but now resolved.

**Pull request merged**